pkg/controllers/node/health/suite_test.go

```diff
@@ -288,34 +288,74 @@ var _ = Describe("Node Health", func() {
 			nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
 			Expect(nodeClaim.DeletionTimestamp).ToNot(BeNil())
 		})
-		It("should ignore unhealthy nodes if more then 20% of the nodes are unhealthy", func() {
+		It("should ignore unhealthy nodes if more then 20% of the nodes are unhealthy in a nodepool", func() {
 			ExpectApplied(ctx, env.Client, nodePool)
-			nodeClaims := []*v1.NodeClaim{}
-			nodes := []*corev1.Node{}
+			nodeClaims, nodes := test.NodeClaimsAndNodes(10, v1.NodeClaim{ObjectMeta: metav1.ObjectMeta{Finalizers: []string{v1.TerminationFinalizer}}})
+			for i := range 3 {
+				nodes[i].Status.Conditions = append(node.Status.Conditions, corev1.NodeCondition{
+					Type:               "BadNode",
+					Status:             corev1.ConditionFalse,
+					LastTransitionTime: metav1.Time{Time: fakeClock.Now()},
+				})
+				nodes[i].Labels[v1.NodePoolLabelKey] = nodePool.Name
+				nodeClaims[i].Labels[v1.NodePoolLabelKey] = nodePool.Name
+			}
 			for i := range 10 {
-				nodeClaim, node = test.NodeClaimAndNode(v1.NodeClaim{ObjectMeta: metav1.ObjectMeta{Finalizers: []string{v1.TerminationFinalizer}}})
-				if i < 3 {
-					node.Status.Conditions = append(node.Status.Conditions, corev1.NodeCondition{
-						Type:               "BadNode",
-						Status:             corev1.ConditionFalse,
-						LastTransitionTime: metav1.Time{Time: fakeClock.Now()},
-					})
-				}
-				node.Labels[v1.NodePoolLabelKey] = nodePool.Name
-				nodeClaim.Labels[v1.NodePoolLabelKey] = nodePool.Name
-				nodeClaims = append(nodeClaims, nodeClaim)
-				nodes = append(nodes, node)
-				ExpectApplied(ctx, env.Client, nodePool, nodeClaim, node)
+				ExpectApplied(ctx, env.Client, nodeClaims[i], nodes[i])
 			}
-
 			fakeClock.Step(60 * time.Minute)
 
-			// Determine to delete unhealthy nodes
+			// Determine if we should delete unhealthy nodes
+			nodeOne := nodes[0]
+			nodeClaimOne := nodeClaims[0]
+			result := ExpectObjectReconciled(ctx, env.Client, healthController, nodeOne)
+			Expect(result.RequeueAfter).To(BeNumerically("~", time.Minute*5, time.Second))
+			nodeClaim = ExpectExists(ctx, env.Client, nodeClaimOne)
+			Expect(nodeClaim.DeletionTimestamp).To(BeNil())
+
+			nodeTwo := nodes[1]
+			nodeClaimTwo := nodeClaims[1]
+			result = ExpectObjectReconciled(ctx, env.Client, healthController, nodeTwo)
+			Expect(result.RequeueAfter).To(BeNumerically("~", time.Minute*5, time.Second))
+			nodeClaim = ExpectExists(ctx, env.Client, nodeClaimTwo)
+			Expect(nodeClaim.DeletionTimestamp).To(BeNil())
+		})
+		It("should ignore unhealthy nodes if more then 20% of the nodes are unhealthy in a cluster", func() {
+			nodeClaims, nodes := test.NodeClaimsAndNodes(10, v1.NodeClaim{ObjectMeta: metav1.ObjectMeta{Finalizers: []string{v1.TerminationFinalizer}}})
 			for i := range 4 {
```

Comment:
nit: Just curious why you would choose 4 here and wouldn't want to just match what you did above with only 3 exceeding the 20% limit